### PR TITLE
Network connections: use raw content when distributing gutenberg->gutenberg

### PR DESCRIPTION
**Pull request by adamsilverstein**
* doesn't check for specific post type gutenberg support, only whether gutenberg is enabled on the remote site

Partially fixes https://github.com/10up/distributor/issues/143

### CODE REVIEW

**Comment by adamsilverstein** [comment]
* supports network connects and external connections. when both original and remote site support gutenberg, raw content is distributed
* stores the gutenberg active status of an external connection when you save it (resave to activate feature for existing connections)
* does not handle subscriptions yet, or post types that may or may not support gutenberg (eg rest not enabled)
* add a new 'dt_the_content' filter applied to final content so plugins can change the default behavior
* will work to get unit tests running

**Comment by barryceelen** [comment]
@adamsilverstein Although I love the explicit nature of the `dt_the_content` filter, in the spirit of frugality one might argue that the content can already be filtered via the `dt_push_post_args` filter.

**Comment by adamsilverstein** [comment]
good point @barryceelen - thanks for the feedback! I also noticed this and later removed the filters in https://github.com/10up/distributor/pull/145/commits/22732adf9696d95807befe2f46672ef4a2c8bec2

I did add a new filter to the subscription updates: https://github.com/10up/distributor/pull/145/files#diff-12f37ea7c389dfd5a6a70cc9992d8935R278

**Comment by adamsilverstein** [comment]
* subscriptions are now supported, although when i tested updating a subscribed post using the default demo post, gutenberg lost the header block. 

It looked fine on the initial distribution, however after an update the header was corrupted. Comparing html, on initial distribution:


After an update:


Note the background style is missing, looking into why.

**Comment by adamsilverstein** [comment]
Subscriptions now work correctly after https://github.com/10up/distributor/pull/145/commits/7b454858da33767d02d58003fb80b2dcd0d9501d. I'd still like to get unit tests running again, although their failure is not related to this PR. Also, the current PR is missing any post_type support - all post types are assumed to support G utenberg

File: includes/rest-api.php [hunk 1]
```diff
@@ -20,11 +20,33 @@ function setup() {
...
+ * @param stdClass        $prepared_post An object representing a single post prepared.
+ * @param WP_REST_Request $request       Request object.
+ *
+ * @return stdClass $prepared_post The filtered post object.
+ */
+function filter_distributor_content( $prepared_post, $request ) {
+
+	if ( \Distributor\Utils\is_using_gutenberg() && isset( $request['distributor_raw_content'] ) ) {
+		$prepared_post->post_content = $request['distributor_raw_content'];
+					error_log( json_encode( $prepared_post, JSON_PRETTY_PRINT ) );
```
Comment (barryceelen) [inline includes/rest-api.php]: @adamsilverstein A lonely error_log is missing its parents and would like to be picked up at Småland!

Comment (tlovett1) [inline includes/rest-api.php]: @adamsilverstein I think this got left in by accident.

Comment (adamsilverstein) [reply includes/rest-api.php]: removed

**Comment by adamsilverstein** [comment]
* added support for post types, via `gutenberg_can_edit_post_type`, needs testing
* realized i need to test/address pull as well

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,6 @@
 language: php
 
 php:
-  - 5.6
```
Comment (adamsilverstein) [inline .travis.yml]: wp-mock doesn't work on 5.6

File: .travis.yml [hunk 3]
```diff
@@ -14,7 +13,7 @@ before_script:
   - composer install
 
 script:
-  - ./vendor/bin/phpcs
+  - ./vendor/bin/phpcs **/*.php *.php
```
Comment (adamsilverstein) [inline .travis.yml]: phpcs complained no files were being scanned

File: tests/php/includes/TestCase.php [hunk 4]
```diff
@@ -2,7 +2,7 @@
 /**
  * We will extend this test case to make WP_Mock set up easier
  */
-class TestCase extends \PHPUnit_Framework_TestCase {
+class TestCase extends \PHPUnit\Framework\TestCase {
```
Comment (adamsilverstein) [inline tests/php/includes/TestCase.php]: changed in latest PHPUnit

**Comment by adamsilverstein** [comment]
* got tests and phpcs running, seeing many phpcs issues - are these already being addressed in a separate issue?

**Comment by barryceelen** [comment]
There is no issue yet for WPCS compliance @adamsilverstein.

**Pull request merged**